pkg/schema/definitions/schema.go

```diff
@@ -0,0 +1,51 @@
+package definitions
+
+import (
+	"time"
+
+	"github.com/rancher/apiserver/pkg/types"
+	"github.com/rancher/wrangler/pkg/schemas"
+	"k8s.io/client-go/discovery"
+)
+
+const (
+	gvkExtensionName    = "x-kubernetes-group-version-kind"
+	gvkExtensionGroup   = "group"
+	gvkExtensionVersion = "version"
+	gvkExtensionKind    = "kind"
+	defaultDuration     = time.Second * 5
```

Comment:
The goal is to provide a sane default for how often we retrieve the openapi doc (which is a large JSON, and l think is reasonably expensive for the api server to produce).

Originally, I tried using a CRD controller that refreshed when a CRD was added/removed, however, the doc isn't immediately updated when a CRD is added, and there did not seem to be any status field or such that would give me an indication of when the aggregated doc would be updated. This solution, while not the best, provides us with a mechanism to ensure that we ask for this value at most once every 5 seconds. 

This being said, I agree that this isn't optimal and am open to other solutions here if you have a suggestion.

Reply:
I do not have a value recommendation as of today.

I am more thinking about the need for a refresh mechanism that is continuously triggered rather than on-demand, but let's get to the bottom of that discussion in the other comment https://github.com/rancher/steve/pull/139/files#r1455052494